premium/offline.go

```diff
@@ -0,0 +1,91 @@
+package premium
+
+import (
+	"crypto/ed25519"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"os"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+type License struct {
+	LicensedTo string    `json:"licensed_to"` // Customers name, e.g. "Acme Inc"
+	IssuedAt   time.Time `json:"issued_at"`
+	ValidFrom  time.Time `json:"valid_from"`
+	ExpiresAt  time.Time `json:"expires_at"`
+}
+
+type LicenseWrapper struct {
+	LicenseBytes []byte `json:"license"`
+	Signature    string `json:"signature"` // crypto
+}
+
+var (
+	ErrInvalidLicenseSignature = errors.New("invalid license signature")
+	ErrLicenseNotValidYet      = errors.New("license not valid yet")
+	ErrLicenseExpired          = errors.New("license expired")
+)
+
+func ValidateLicense(logger zerolog.Logger, licenseFile string) error {
+	licenseContents, err := os.ReadFile(licenseFile)
+	if err != nil {
+		return err
+	}
+
+	l, err := UnpackLicense(licenseContents)
+	if err != nil {
+		return err
+	}
+
+	return l.IsValid(logger)
+}
+
+func UnpackLicense(lic []byte) (*License, error) {
+	const publicKey = "96e4749b7550d33bd776cb7fb056d74cb16ed69dfe8e59c16e8c2500a94162b1"
```

Comment:
move this to separate file & use go:embed

Reply:
Done in [`07fd136` (#1436)](https://github.com/cloudquery/plugin-sdk/pull/1436/commits/07fd13685b0a0fd9ea82bd28ae93d69ea3d03c06)